app/src/main/java/org/oppia/android/app/profile/ProfileChooserFragmentPresenter.kt

```diff
@@ -251,4 +253,10 @@ class ProfileChooserFragmentPresenter @Inject constructor(
       oppiaClock.getCurrentTimeMs(), eventContext = oppiaLogger.createOpenProfileChooserContext()
     )
   }
+
+  private fun updateLearnerIdIfAbsent(profile: Profile) {
+    if (profile.learnerId.isNullOrEmpty()) {
+      profileManagementController.updateLearnerId(profile.id)
+    }
```

Comment:
It would probably be simpler to just do this during profile retrieval, but more thought is needed.